docs/docs/usage/scaladoc/settings.md

```diff
@@ -134,3 +142,38 @@ Example JSON file:
 }
 ```
 
+##### -snippet-compiler
+
+Snippet compiler arguments provide a way to configure snippet checking.
+
+This setting accepts a list of arguments in the format:
+args := arg{,arg}
+arg := [path=]flag
+where the path is a prefix of source paths to members to which argument should be set.
+
+If the path is not present, the argument will be used as the default for all unmatched paths.
+
+Available flags:
+compile - Enables snippet checking.
+nocompile - Disables snippet checking.
+fail - Enables snippet checking, asserts that snippet doesn't compile.
+
+The fail flag comes in handy for snippets that present that some action would eventually fail during compilation, e. g. [Opaques page](docs/reference/other-new-features/opaques.html)
+
+Example usage:
+
+`-snippet-compiler:my/path/nc=nocompile,my/path/f=fail,compile`
+
+Which means:
+
+all snippets in files under directory `my/path/nc` should not be compiled at all
+all snippets in files under directory `my/path/f` should fail during compilation
+all other snippets should compile successfully
+
+##### -Ysnippet-compiler-debug
+
+Setting this option causes snippet compiler to print snippet as it is compiled (after wrapping).
+
+##### -Ydocument-synthetic-types
+
+Documents intrinsic types e. g. Any, Nothing. Setting is useful only for stdlib
```

Comment:
What does *to document* mean here? Does it annotate them?

Reply:
Will be `Attach pages with documentation of the intrinsic types e. g. Any, Nothing to the docs. Setting is useful only for stdlib.` ok? The setting is supposed to add few pages to the `API` tree in resulting scaladoc. This is done this way, because there are no tasty files we could create for these types to pass to scaladoc, but for all other users this settings should not be enabled.

Reply:
I would prefer
`Include pages providing documentation for the intrinsic types (e. g. `Any`, `Nothing`) to the docs. Setting is useful only for stdlib.

Reply:
Thank you for explaining. I think you can include what you mentioned in this reply into the doc

Reply:
Final version:
`Include pages providing documentation for the intrinsic types (e. g. Any, Nothing) to the docs. The setting is useful only for stdlib.
Scaladoc for Scala 3 works for tasty files, but we cannot provide them for intrinsic types since they are embedded in the compiler.
All other users should not concern with this setting.` 

What do you think?

Reply:
> Include pages providing documentation for the intrinsic types (e. g. Any, Nothing) to the docs. The setting is useful only for stdlib because scaladoc for Scala 3 relies on TASTy files, but we cannot provide them for intrinsic types since they are embedded in the compiler. 
All other users should not concern with this setting.

Reply:
Changed, thanks :D